builtin/credential/aws/path_role.go

```diff
@@ -41,14 +54,32 @@ that it must match the IAM role ARN specified by this parameter.
 The value is prefix-matched (as though it were a glob ending in
 '*').  The configured IAM user or EC2 instance role must be allowed
 to execute the 'iam:GetInstanceProfile' action if this is
-specified.`,
+specified. This is only checked when auth_type is
+ec2.`,
 			},
 			"bound_iam_instance_profile_arn": {
 				Type: framework.TypeString,
 				Description: `If set, defines a constraint on the EC2 instances to be associated
 with an IAM instance profile ARN which has a prefix that matches
 the value specified by this parameter. The value is prefix-matched
-(as though it were a glob ending in '*').`,
+(as though it were a glob ending in '*'). This is only checked when
+auth_type is ec2.`,
+			},
+			"inferred_entity_type": {
+				Type: framework.TypeString,
+				Description: `When auth_type is iam, the
+AWS entity type to infer from the authenticated principal. The only supported
+value is ec2_instance, which will extract the EC2 instance ID from the
+authenticated role and apply restrictions specific to EC2 instances (such as
+role_tag). The configured EC2 client must be able to find the inferred instance
+ID in the results, and the instance must be running. If unable to
+determine the EC2 instance ID or unable to find the EC2 instance ID
+among running instances, then authentication will fail.`,
+			},
+			"inferred_aws_region": {
```

Comment:
When `inferred_entity_type` is set, does having a default `inferred_aws_region` makes sense?